src/corona_hakab_model/medical_state_manager.py

```diff
@@ -11,11 +11,18 @@ class MedicalStateManager:
     Manages the medical state
     """
 
-    def __init__(self, sim_manager: "manager.SimulationManager"):
-        self.manager = sim_manager
+    def __init__(self, medical_state_machine: MedicalStateMachine):
+        self.medical_state_machine = medical_state_machine
         self.pending_transfers = PendingTransfers()
 
-    def step(self, new_sick: List[Agent]):
+    def step(self, new_sick: List[Agent], update_only_medical_state: bool = False):
+        """
+
+        :param new_sick: List of new agents that got sick
+        :param update_only_medical_state: Defaults to False. If True, update agent.medical_state directly.
+                                          If False it also updates agent.manager.
```

Comment:
It seems that you want to access the manager through the agents, which looks kind of twisted to me.
In the flow of 'update_only_medical_state' just ignore the manager and leave it as None